From 3410059dea2a88f9606a5160a890408620c94a11 Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Wed, 24 Jun 2026 07:41:00 +0000
Subject: [PATCH] wifi: brcmfmac: validate bsscfg indices in IF events

jira VULN-185617
cve CVE-2026-43110
commit-author Pengpeng Hou <pengpeng@iscas.ac.cn>
commit 304950a467d83678bd0b0f46331882e2ac23b12d
upstream-diff |
	Use `brcmf_err` instead of `bphy_err` due to missing
	dcb1471bc6d05 ("brcmfmac: use bphy_err() in all wiphy-related code")

brcmf_fweh_handle_if_event() validates the firmware-provided interface
index before it touches drvr->iflist[], but it still uses the raw
bsscfgidx field as an array index without a matching range check.

Reject IF events whose bsscfg index does not fit in drvr->iflist[]
before indexing the interface array.

	Signed-off-by: Pengpeng Hou <pengpeng@iscas.ac.cn>
	Acked-by: Arend van Spriel <arend.vanspriel@broadcom.com>
Link: https://patch.msgid.link/20260323074551.93530-1-pengpeng@iscas.ac.cn
[add missing wifi prefix]
	Signed-off-by: Johannes Berg <johannes.berg@intel.com>
(cherry picked from commit 304950a467d83678bd0b0f46331882e2ac23b12d)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/fweh.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fweh.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fweh.c
index ef72baf6dd96..f5bd7ab1084b 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fweh.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/fweh.c
@@ -161,6 +161,10 @@ static void brcmf_fweh_handle_if_event(struct brcmf_pub *drvr,
 		brcmf_err("invalid interface index: %u\n", ifevent->ifidx);
 		return;
 	}
+	if (ifevent->bsscfgidx >= BRCMF_MAX_IFS) {
+		brcmf_err("invalid bsscfg index: %u\n", ifevent->bsscfgidx);
+		return;
+	}
 
 	ifp = drvr->iflist[ifevent->bsscfgidx];
 
-- 
2.52.0

